site/docs/hosting/firebase.md

```diff
@@ -0,0 +1,185 @@
+# Hosting: Firebase Functions
+
+This tutorial will guide you through the process of deploying your bot to [Firebase Functions](https://firebase.google.com/docs/functions).
+
+## Prerequisites
+
+To follow along, you will need to have a Google account.
+If you don't already have one, you can create it [here](https://accounts.google.com/signup).
+
+## Setup
+
+This section guides you through the setup process.
+If you need more detailed explanations on each step that you'll take, refer to the [official Firebase documentation](https://firebase.google.com/docs/functions/get-started).
+
+### Creating a Firebase Project
+
+1. Go to the [Firebase console](https://console.firebase.google.com/) and click **Add Project**.
+2. If prompted, review and accept the Firebase terms.
+3. Click **Continue**.
+4. Decide on whether you want to share analytics or not.
+5. Click **Create Project**.
+
+### Setting Things Up
+
+To write functions and deploy them to the Firebase Functions runtime, you'll need to set up a Node.js environment and install the Firebase CLI.
+
+> It's important to note that only the Node.js versions 14, 16, and 18 are currently supported by Firebase Functions.
+> For more on the supported Node.js versions, refer to [here](https://firebase.google.com/docs/functions/manage-functions#set_nodejs_version).
+
+Once you have Node.js and NPM installed, install Firebase CLI globally:
+
+```sh
+npm install -g firebase-tools
+```
+
+### Initializing the Project
+
+1. Run `firebase login` to open your browser and authenticate Firebase CLI with your account.
+2. `cd` into your project's directory.
+3. Run `firebase init functions`, and type `y` when asked whether you want to initialize a new codebase.
+4. Choose `use existing project` and select the project you created in Step 1.
+5. The CLI gives you two options for language support:
+   - JavaScript
+   - TypeScript
+
+   For this tutorial, you can select either.
+
```

Comment:
So
```suggestion
```
?

Reply:
I agree with you. I think we're done here. This tutorial is an excellent piece of work and I don't feel like dragging it out further. I'll just commit this now and let @roj1512 revert in case I misunderstood :)

Reply:
wonderfull, thank you :)